src/Persistence/MassTransit.Azure.Storage/AzureStorage/MessageData/AzureStorageMessageDataRepository.cs

```diff
@@ -151,5 +151,25 @@ static async Task SetBlobExpiration(BlobBaseClient blob, TimeSpan? timeToLive)
                 await blob.SetMetadataAsync(metadata).ConfigureAwait(false);
             }
         }
+
+        public async Task Delete(Uri address, CancellationToken cancellationToken = default)
+        {
+            if (address == null)
+                return;
+
+            var blobName = new BlobUriBuilder(address).BlobName;
+            var blob = _container.GetBlobClient(blobName);
+
+            try
+            {
+                LogContext.Debug?.Log("DELETE Message Data: {Address} ({Blob})", address, blobName);
+
+                await blob.DeleteAsync(DeleteSnapshotsOption.IncludeSnapshots, default, cancellationToken).ConfigureAwait(false);
+            }
+            catch (RequestFailedException exception)
+            {
+                throw new MessageDataNotFoundException(address, new MessageDataException($"MessageData content not found: {blob.BlobContainerName}/{blob.Name}", exception));
```

Comment:
Trying to delete something that doesn't exist shouldn't be a fail, operations should be idempotent.

Reply:
perfect thank you.